### ISSUE TITLE
boost-test: linking to static library fails

### ISSUE THREAD

**Issue by madebr** [bincrafters/community#167]
### Description of Problem, Request, or Question
Linking to static boost-test library fails on Linux/Windows.

Error is:
```
main.obj : error LNK2019: unresolved external symbol "int __cdecl boost::unit_test::unit_test_main(bool (__cdecl*)(void),int,char * * const)" (?unit_test_main@unit_test@boost@@YAHP6A_NXZHQEAPEAD@Z) referenced in function main [C:\issues\conan_boost_test_2\build\issue.vcxproj]
```

### Package Details (Include if Applicable)
* Package Name/Version: **boost-test/1.66.0**
* Operating System: **Windows**
* Compiler+version: **Visual Studio 2017**

### Steps to reproduce (Include if Applicable)
project: https://github.com/madebr/issues/tree/master/conan_boost_test_2
appveyor: https://ci.appveyor.com/project/madebr/issues/build/job/2bgbecoufhsaynvr
travis: https://travis-ci.org/madebr/issues/jobs/351413461

I followed the steps at http://www.boost.org/doc/libs/1_66_0/libs/test/doc/html/boost_test/adv_scenarios/static_lib_customizations/entry_point.html

**Comment by madebr** [comment]
Omitting the `main` function and `boost::unit_test::unit_test_main` function call causes the following error on Windows alone:
```
libboost_prg_exec_monitor.lib(cpp_main.obj) : error LNK2019: unresolved external symbol "int __cdecl cpp_main(int,char * * const)" (?cpp_main@@YAHHQEAPEAD@Z) referenced in function main [C:\issues\conan_boost_test_2\build\issue.vcxproj]
```
appveyor: https://ci.appveyor.com/project/madebr/issues/build/job/2bgbecoufhsaynvr

**Comment by solvingj** [comment]
@madebr i think this might be resolved in 1.67.0+.  Can you give a try and let us know?

**Comment by madebr** [comment]
I haven't seen this or similar errors since more then a year. So closing this issue.
